Simplenote/NoticePresenter.swift

```diff
@@ -0,0 +1,170 @@
+import UIKit
+
+class NoticePresenter: KeyboardObservable {
+
+    // MARK: Properties
+    //
+    private var containerView = PassthruView(frame: .zero)
+    var noticeView: NoticeView?
+    private var noticeVariableConstraint: NSLayoutConstraint?
+
+    var isPresenting: Bool {
+        noticeView != nil
+    }
+    private var keyboardHeight: CGFloat = .zero
+    private var keyboardFloats: Bool = false
+    private var keyboardNotificationTokens: [Any]?
+    var timer: Timer?
+
+
+    // MARK: Lifecycle
+    //
+    deinit {
+        stopListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting/Dismissing Methods
+    //
+    func presentNoticeView(_ noticeView: NoticeView, completion: @escaping () -> Void) {
+        guard let keyWindow = getKeyWindow() else {
+            return
+        }
+        self.noticeView = noticeView
+
+        containerView.addSubview(noticeView)
+        keyWindow.addFillingSubview(containerView)
+
+        noticeVariableConstraint = noticeView.topAnchor.constraint(equalTo: containerView.bottomAnchor)
+        noticeVariableConstraint?.isActive = true
+        noticeView.centerXAnchor.constraint(equalTo: containerView.centerXAnchor).isActive = true
+        containerView.layoutIfNeeded()
+
+        displayNotificationView() {
+            completion()
+        }
+    }
+
+    private func displayNotificationView(completion: @escaping () -> Void) {
+        prepareConstraintToDisplayNotice()
+
+        UIView.animate(withDuration: UIKitConstants.animationLongDuration) {
+            self.containerView.layoutIfNeeded()
+        } completion: { (_) in
+            completion()
+        }
+    }
+
+    private func makeBottomConstraintConstant() -> CGFloat {
```

Comment:
```suggestion
    private var noticeBottomPadding: CGFloat {
```

Typically all of the `make*` APIs would return objects (but in this particular case it's a calculation)